src/query/context.hpp

```diff
@@ -66,6 +66,7 @@ struct ExecutionContext {
   SymbolTable symbol_table;
   EvaluationContext evaluation_context;
   std::atomic<bool> *is_shutting_down{nullptr};
+  std::atomic<bool> *aborted_by_user{nullptr};
```

Comment:
Because we have to sync this flag between two threads. Example:
Thread A runs long-running query.
Thread B runs kill transaction and sets aborted_by_user to true.
In order that thread A sees change made from thread B, it needs to pick up the fresh value, hence atomic.